metpy/calc/kinematics.py

```diff
@@ -978,9 +860,9 @@ def q_vector(u, v, temperature, pressure, dx, dy, static_stability=1):
     static_stability
 
     """
-    dudy, dudx = gradient(u, deltas=(dy, dx), axis=(-2, -1))
```

Comment:
This is funny because our `gradient` doesn't take an `axis` kwarg until your changes later--I'm pretty sure you need to specify axis in order to work with 3D arrays.

Reply:
I noticed that too...should have paid more attention when working on the Q-vectors. :smile: